bpdm-orchestrator/src/main/kotlin/org/eclipse/tractusx/bpdm/orchestrator/service/TimeoutProcessBatchService.kt

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2021,2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.eclipse.tractusx.bpdm.orchestrator.service
+
+import jakarta.persistence.EntityManager
+import mu.KotlinLogging
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Service
+
+@Service
+class TimeoutProcessBatchService(
+    private val goldenRecordTaskService: GoldenRecordTaskService,
+    private val entityManager: EntityManager
+) {
+    private val logger = KotlinLogging.logger { }
+
+    @Scheduled(cron = "\${bpdm.task.timeoutCheckCron}")
+    fun processForTimeouts() {
+        try {
+            logger.debug { "Checking for timeouts" }
+            processTimeouts(goldenRecordTaskService::processPendingTimeouts)
+            processTimeouts(goldenRecordTaskService::processRetentionTimeouts)
+        } catch (err: RuntimeException) {
+            logger.error(err) { "Error checking for timeouts" }
+        }
```

Comment:
Here we could have a final log for the total amount of processed tasks and that the scheduling is finished. When testing I was not sure whether the schedule was still going or already finished

Reply:
Done